app/controllers/users_controller.rb

```diff
@@ -0,0 +1,42 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      create_user_roles
+      # TODO: This should redirect to the Show page of that user
+      redirect_to root_path, flash: { success: t(".success") }
+    else
+      flash.now[:error] = @user.errors.full_messages.to_sentence
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def user_params
+    params.expect(user: %i[username password password_confirmation])
+  end
+
+  def create_user_roles
+    return unless params[:user]&.key?(:roles)
```

Comment:
Instead of manually using the `params` hash here, do the following:

1) Change your `user_params` to allow a `roles` array through: `params.expect(user: [:username, :password, :password_confirmation, { roles: [] }])`.
2) Use `user_params[:roles] || []` in your line below
3) Remove this line (you already default to `[]` if roles is nil in the next line.